components/AvalancheCenterSelector.tsx

```diff
@@ -44,55 +45,84 @@ const center_idsByType: SectionListData<string>[] = [
-      </View>
+      <Box px="2" pt="2">
+        <Box bg="white" borderWidth="2" borderRadius="8" borderColor="light.200" p="4">
+          <VStack space="2">
+            <HStack justifyContent="space-between" pt="4">
+              <NBText color="light.400">{`Could not fetch data for ${center_id}: ${error?.message}.`}</NBText>
+            </HStack>
+          </VStack>
+        </Box>
+      </Box>
     );
   }
   if (!avalancheCenter) {
     return (
-      <View style={styles.item}>
-        <Text>{`No metadata found for ${center_id}.`}</Text>
-      </View>
+      <Box px="2" pt="2">
+        <Box bg="white" borderWidth="2" borderRadius="8" borderColor="light.200" p="4">
+          <VStack space="2">
+            <HStack justifyContent="space-between" pt="4">
+              <NBText color="light.400">{`No metadata found for ${center_id}.`}</NBText>
+            </HStack>
+          </VStack>
+        </Box>
+      </Box>
     );
   }
 
   return (
-    <TouchableOpacity
-      style={styles.item}
-      onPress={() => {
-        set_avalanche_center(center_id);
-        // We need to clear navigation state to force all screens from the
-        // previous avalanche center selection to unmount
-        navigation.reset({
-          index: 0,
-          routes: [{name: 'Home'}],
-        });
-      }}>
-      <AvalancheCenterLogo style={styles.logo} center_id={center_id} />
-      <Text style={{textAlignVertical: 'center'}}>{avalancheCenter.name}</Text>
-    </TouchableOpacity>
+    <Box px="2" pt="2">
+      <TouchableOpacity
+        style={styles.item}
+        onPress={() => {
+          onPress(center_id);
+        }}>
+        <Box bg={selected ? 'red.400' : 'white'} borderWidth="2" borderRadius="8" borderColor="light.200" p="4">
+          <VStack space="2">
+            <HStack justifyContent="space-between" pt="4">
+              <AvalancheCenterLogo style={styles.logo} center_id={center_id} />
+              <NBText color="light.400">{avalancheCenter.name}</NBText>
+            </HStack>
+          </VStack>
+        </Box>
+      </TouchableOpacity>
+    </Box>
   );
 };
 
-export const AvalancheCenterSelector = ({setAvalancheCenter}) => {
+export const AvalancheCenterSelector = ({currentCenterId, setAvalancheCenter}) => {
+  const navigation = useNavigation<TabNavigationProps>();
   return (
     <SectionList
       style={styles.container}
       sections={center_idsByType}
       keyExtractor={item => item}
-      renderItem={({item}) => <AvalancheCenterCard center_id={item} set_avalanche_center={setAvalancheCenter} />}
+      renderItem={({item}) => (
+        <AvalancheCenterCard
+          center_id={item}
+          selected={item === currentCenterId}
+          onPress={(center_id: string) => {
+            setAvalancheCenter(center_id);
+            // We need to clear navigation state to force all screens from the
+            // previous avalanche center selection to unmount
+            navigation.reset({
```

Comment:
We can probably nix this navigation now - and just have it update the state, let the user pop out of the menu as they need to

Reply:
that seems reasonable to me